src/Azure.Functions.Cli/Actions/LocalActions/CreateFunctionAction.cs

```diff
@@ -19,18 +25,100 @@ internal class CreateFunctionAction : BaseAction
     {
         private readonly ITemplatesManager _templatesManager;
         private readonly ISecretsManager _secretsManager;
+        private readonly ISettings _settings;
 
         public string Language { get; set; }
         public string TemplateName { get; set; }
         public string FunctionName { get; set; }
         public bool Csx { get; set; }
 
-        public CreateFunctionAction(ITemplatesManager templatesManager, ISecretsManager secretsManager)
+        public CreateFunctionAction(ITemplatesManager templatesManager, ISecretsManager secretsManager, ISettings settings)
         {
             _templatesManager = templatesManager;
             _secretsManager = secretsManager;
+            _settings = settings;
         }
 
+        private async Task<(bool, string)> CallAzCommandAsync(string commandName)
+        {
+            var windowscmd = new StringBuilder();
+            windowscmd.Append("/c az ").Append(commandName);
+
+            var az = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                        ? new Executable("cmd", windowscmd.ToString())
+                        : new Executable("az", commandName);
+
+            var stdout = new StringBuilder();
+            var stderr = new StringBuilder();
+            var exitCode = await az.RunAsync(o => stdout.AppendLine(o), e => stderr.AppendLine(e));
+            if (exitCode == 0)
+            {
+                return (true, (stdout.ToString().Trim(' ', '\n', '\r', '"')));
+            }
+            else
+            {
+                return (false, (stderr.ToString().Trim(' ', '\n', '\r'))); // "Unable to connect to Azure. Make sure you have the `az` CLI installed and logged in and try again";
+            }
+        }
+
+        private async Task CreateWebJobsStorageKeyAsync()
+        {
+            ColoredConsole.Write("Please enter the name of your resource group for your storage account: ");
+            string resource_group = Console.ReadLine();
+            
+            ColoredConsole.Write("Please enter the name of your storage account: ");
+            string storage_account = Console.ReadLine();
+
+            //verify if the resource group and account exists and get the details
+            //az storage account show -g MyResourceGroup -n MyStorageAccount
+            var command = new StringBuilder();
+            command.Append("storage account keys list -g ").Append(resource_group).Append(" -n ").Append(storage_account).Append(" --output json");
```

Comment:
You could use [string interpolation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated) here. There should be examples throughout the project.

Something like `var command = $"storage account keys list -g {resource_group} -n {storage_account} --output {json}"`